R/paginate.R

```diff
@@ -155,31 +170,24 @@ paginate_req_perform <- function(req,
 
 #' @export
 #' @rdname paginate_req_perform
-paginate_next_request <- function(resp, req, parsed) {
-  check_response(resp)
+paginate_next_request <- function(req, parsed) {
   check_request(req)
   check_has_pagination_policy(req)
 
   next_request <- req$policies$paginate$next_request
   next_request(
-    resp = resp,
     req = req,
     parsed = parsed
   )
 }
 
-#' @param next_url A function that extracts the url to the next page from the
-#'   [response] and the `body`.
 #' @rdname req_paginate
 #' @export
 req_paginate_next_url <- function(req,
-                                  next_url,
-                                  parse_resp = NULL,
+                                  parse_resp,
                                   n_pages = NULL) {
-  check_function2(next_url, args = c("resp", "parsed"))
-
-  next_request <- function(req, resp, parsed) {
-    next_url <- next_url(resp, parsed)
+  next_request <- function(req, parsed) {
+    next_url <- parsed[["next_url"]]
```

Comment:
The question is how/where to document it best, that `parse_resp()` needs to return a list that has field `next_url` for `req_paginate_next_url()`?